components/sup/src/manager/sup_watcher.rs

```diff
@@ -0,0 +1,55 @@
+//! Watcher interface implementation for Habitat Supervisor.
+
+use notify::{poll::PollWatcher,
+             DebouncedEvent,
+             RecommendedWatcher,
+             RecursiveMode,
+             Result,
+             Watcher};
+use std::{env,
+          path::Path,
+          sync::mpsc::Sender,
+          time::Duration};
+
+pub enum SupWatcher {
+    Native(RecommendedWatcher),
+    Fallback(PollWatcher),
+}
+
+impl Watcher for SupWatcher {
+    fn new_raw(tx: Sender<notify::RawEvent>) -> Result<Self> {
+        if let Ok(arch_type) = env::var("HAB_STUDIO_HOST_ARCH") {
```

Comment:
This might be a more idiomatic way of doing this given we already have platform architecture detection capabilities in the codebase.
```rust
// The habitat_core::package::target contains constants for all of habitat's
// supported platforms.
use habitat_core::package::target::{PackageTarget, AARCH64_DARWIN};
// This will replace the HAB_STUDIO_HOST_ARCH environment variable with the default current arch
// if the current value is invalid
let target = PackageTarget::from_str(env::var("HAB_STUDIO_HOST_ARCH").unwrap_or_default()).unwrap_or(PackageTarget::active_target());
// If the studio host target is AARCH64_DARWIN 
if target == AARCH64_DARWIN {
    // Use poll watcher
} else {
    // Use recommended watcher
}
```

Reply:
This won't work because the following yields "aarch64-macos", not "aarch64-darwin":
env::set_var(STUDIO_HOST_ARCH_ENVVAR, format!("{}-{}", std::env::consts::ARCH,std::env::consts::OS)
I thought of adding this to habitat_core::package target.rs
    ("aarch64-macos", AARCH64_Macos, AARCH64_MACOS, "aarch64", "macos");
 But I think it would add confusion.

Reply:
The lookup fails however with this approach.
I did try adding this line to the target.rs for PackageTarget:
("aarch64-macos", AARCH64_Darwin, AARCH64_DARWIN, "aarch64", "macos");
and rebuilding.
But I would always get back the default x86_64-linux.  
So I left this code as is until we can talk about it.

Reply:
When we start supervisor inside of the studio, won't PackageTarget::active_target() resolve to x86_64-linux, not aarch64-darwin?

Reply:
When setting `HAB_STUDIO_SECRET_HAB_STUDIO_HOST_ARCH` you would need to set it to `*PackageTarget::active_target()`. 

Yes, `PackageTarget::active_target()` will always be x86 in the studio however in the above logic it would only be used as a fallback if there is no `HAB_STUDIO_HOST_ARCH` set.